### PR TITLE
Profile page and Create Account

### PR DESCRIPTION
**Pull request by lindakumah**

### CODE REVIEW

File: cypress/integration/tests/AccountCreation/amali-create.account.spec.js [hunk 1]
```diff
@@ -6,65 +6,57 @@ import { baseUrl } from 'cypress/fixtures/shared/environment.js'
...
-      cy.get('#signup_password').as('passwordField')
-      cy.get('@passwordField').type(password)
-      cy.get('#confirm_password').as('confirmPasswordField')
-      cy.get('@confirmPasswordField').type(confirm_password)
+    it('signs up with valid email and password', () => {
+      cy.location().should((loc) => {
+        expect(loc.pathname).to.eq('/sign-up')
+      })
+      cy.get('#signup_password').type(newuser().password)
+      cy.get('#confirm_password').type(newuser().confirm_password)
```
Comment (JerryHukpati) [inline cypress/integration/tests/AccountCreation/amali-create.account.spec.js:35]: no need to use newuser().password etc. when you have already destructured the values in the newuser function

File: cypress/integration/tests/AccountCreation/amali-create.account.spec.js [hunk 2]
```diff
@@ -6,65 +6,57 @@ import { baseUrl } from 'cypress/fixtures/shared/environment.js'
...
-    //   cy.url().should('include', '/unverify-email')
-    // })
+    it('Signs up with a mismatched password', () => {
+      cy.location().should((loc) => {
+        expect(loc.pathname).to.eq('/sign-up')
+      })
+      cy.get('#signup_password').type(newuser().password)
+      cy.get('#confirm_password').type(newuser().confirm_password2)
+      cy.get('#submit_button').click()
+      cy.get('.v-messages__message').its('length').should('be.gt', 0)
```
Comment (JerryHukpati) [inline cypress/integration/tests/AccountCreation/amali-create.account.spec.js:59]: all dynamic classes should be replaced with static id's/datacy's. speak with the developers about this

**Review by JerryHukpati — CHANGES_REQUESTED** [review]
add device names to your describes to differentiate desktop tests from mobile

File: cypress/integration/tests/AccountCreation/amali-create.account.spec.js [hunk 3]
```diff
@@ -6,65 +6,57 @@ import { baseUrl } from 'cypress/fixtures/shared/environment.js'
...
 
     before(() => {
       cy.viewport(w, h)
-    
-      cy.visit(targetUrl)
-      // cy.checkEmail(email)
-      cy.continueWithNonExistingEmail()
-      // cy.clearCookies({ domain: null })
-    
+      cy.clearCookies({ domain: null })
```
[the remark this answers is not in the document]
Comment (lindakumah) [reply cypress/integration/tests/AccountCreation/amali-create.account.spec.js]: Resolved

File: cypress/integration/tests/AccountCreation/amali-create.account.spec.js [hunk 4]
```diff
@@ -6,65 +6,57 @@ import { baseUrl } from 'cypress/fixtures/shared/environment.js'
...
-
-    it('check sign up with valid email and password', () => {
-      cy.get('#signup_email').as('emailField')
-      cy.get('@emailField').should('have.value', email)
-      cy.get('@emailField').clear().type(email)
-      cy.get('#signup_password').as('passwordField')
-      cy.get('@passwordField').type(password)
-      cy.get('#confirm_password').as('confirmPasswordField')
-      cy.get('@confirmPasswordField').type(confirm_password)
+    it('signs up with valid email and password', () => {
```
[the remark this answers is not in the document]
Comment (lindakumah) [reply cypress/integration/tests/AccountCreation/amali-create.account.spec.js]: Resolved

File: cypress/integration/tests/AccountCreation/amali-create.account.spec.js [hunk 5]
```diff
@@ -6,65 +6,57 @@ import { baseUrl } from 'cypress/fixtures/shared/environment.js'
 
 import { newuser } from 'cypress/fixtures/amali-create.account.data'
```
[the remark this answers is not in the document]
Comment (lindakumah) [reply cypress/integration/tests/AccountCreation/amali-create.account.spec.js]: Resolved

File: cypress/integration/tests/AccountCreation/amali-create.account.spec.js [hunk 6]
```diff
@@ -6,65 +6,57 @@ import { baseUrl } from 'cypress/fixtures/shared/environment.js'
...
-    //   cy.get('@emailField').should('have.value', email)
-    //   cy.get('@emailField').clear().type(email)
-    //   cy.get('#signup_password').as('passwordField')
-    //   cy.get('@passwordField').type(password)
-    //   cy.get('#confirm_password').as('confirmPasswordField')
-    //   cy.get('@confirmPasswordField').type(confirm_password)
-    //   cy.get('#submit_button').click()
-    //   cy.url().should('include', '/unverify-email')
-    // })
+    it('Signs up with a mismatched password', () => {
```
[the remark this answers is not in the document]
Comment (lindakumah) [reply cypress/integration/tests/AccountCreation/amali-create.account.spec.js]: Resolved

File: cypress/integration/tests/Profile/amali-userprofile.spec.js [hunk 7]
```diff
@@ -62,20 +47,22 @@ const targetUrl = `${baseUrl}`
       cy.get('a[href*="/profile"]').click()
     })
 
-    it('navigates User to the Help and Support page', () => {
-      cy.get('a[href*="/help-and-support"]').click()
+    it('takes User to the Edit profile page', () => {
```
[the remark this answers is not in the document]
Comment (lindakumah) [reply cypress/integration/tests/Profile/amali-userprofile.spec.js]: Done!

File: cypress/integration/tests/EditProfile/amali-editprofile.spec.js [hunk 8]
```diff
@@ -166,115 +180,102 @@ defaultDevices.map((device) => {
...
 
       //submit edit form
-      cy.get('button').contains('Save').click()
-      cy.url().should('include', '/profile')
+      cy.get('#btnUpdateProfile').click()
+      cy.url().should('include', '/profile?redirect=%2F')
     })
   })
-})
+// })
```
Comment (JerryHukpati) [inline cypress/integration/tests/EditProfile/amali-editprofile.spec.js:281]: remove comment

File: cypress/integration/tests/EditProfile/amali-editprofile.spec.js [hunk 9]
```diff
@@ -166,115 +180,102 @@ defaultDevices.map((device) => {
...
-        .type(userData.address[randomIndex(userData.address)])
-      cy.get('[id=input-111]')
-        .as('postal-code')
-        .type(userData.postal_code[randomIndex(userData.postal_code)])
-      cy.get('[id=input-127]').as('city').type(userData.city[randomIndex(userData.city)])
+      cy.get('[data-cy="postalCode"]').as('postal-code').type(postalCode)
+
+      cy.get('[data-cy="city"]').as('cityField').type(city)
 
       // select country
```
Comment (JerryHukpati) [inline cypress/integration/tests/EditProfile/amali-editprofile.spec.js:258]: remove comment

File: cypress/integration/tests/EditProfile/amali-editprofile.spec.js [hunk 10]
```diff
@@ -166,115 +180,102 @@ defaultDevices.map((device) => {
...
+      // cy.get('[data-cy="country"]').then((dropdown) => {
+      //   cy.wrap(dropdown).click()
+      //   cy.get('[id="list-152"]')
+      //     .children()
+      //     .then(() => {
+      //       cy.get('. v-list-item__title')
+      //         .contains(userData.country)
+      //         .click()
+      //     })
+      // })
```
Comment (JerryHukpati) [inline cypress/integration/tests/EditProfile/amali-editprofile.spec.js:268]: remove comment

File: cypress/integration/tests/EditProfile/amali-editprofile.spec.js [hunk 11]
```diff
@@ -166,115 +180,102 @@ defaultDevices.map((device) => {
...
-        cy.selectYear(2022)
-        cy.selectMonth(2)
-        cy.selectDay(15)
-      })
+      // cy.get('[data-cy="dob"]').click()
+      // cy.get('.v-picker--date').then(() => {
+      //   cy.selectYear(2022)
+      //   cy.selectMonth(2)
+      //   cy.selectDay(15)
+      // })
```
Comment (JerryHukpati) [inline cypress/integration/tests/EditProfile/amali-editprofile.spec.js:242]: remove comment

File: cypress/integration/tests/EditProfile/amali-editprofile.spec.js [hunk 12]
```diff
@@ -166,115 +180,102 @@ defaultDevices.map((device) => {
...
-      //getPrevious
-      cy.get('[aria-label=Previous month]')
+      cy.get('[data-cy="mobile"]').as('mobileField').type(mobile)
 
-      //getNext Button
-      cy.get('[aria-label=Next month]')
+      cy.get('[data-cy="otherEmail"]').as('alt-email').type(other_email)
 
-      cy.get('.v-picker--date').select([5])
+      cy.get('[data-cy="address"]').as('addressField').type(address)
```
Comment (JerryHukpati) [inline cypress/integration/tests/EditProfile/amali-editprofile.spec.js:252]: why are you chaining these in an alias if won't need it later?

File: cypress/integration/tests/EditProfile/amali-editprofile.spec.js [hunk 13]
```diff
@@ -166,115 +180,102 @@ defaultDevices.map((device) => {
...
-      cy.get('[id=input-124]')
-        .as('address')
-        .type(userData.address[randomIndex(userData.address)])
-      cy.get('[id=input-111]')
-        .as('postal-code')
-        .type(userData.postal_code[randomIndex(userData.postal_code)])
-      cy.get('[id=input-127]').as('city').type(userData.city[randomIndex(userData.city)])
+      cy.get('[data-cy="postalCode"]').as('postal-code').type(postalCode)
+
+      cy.get('[data-cy="city"]').as('cityField').type(city)
```
Comment (JerryHukpati) [inline cypress/integration/tests/EditProfile/amali-editprofile.spec.js:256]: refer to earlier comment on unused aliases

File: cypress/integration/tests/EditProfile/amali-editprofile.spec.js [hunk 14]
```diff
@@ -166,115 +180,102 @@ defaultDevices.map((device) => {
...
         .invoke('val')
         .then(($value) => {
           cy.get('@whatsapp').should('have.value', $value)
         })
-    })
+        cy.get('#btnUpdateProfile').click()
+
+      })
+
+
```
Comment (JerryHukpati) [inline cypress/integration/tests/EditProfile/amali-editprofile.spec.js:224]: remove empty lines

File: cypress/integration/tests/EditProfile/amali-editprofile.spec.js [hunk 15]
```diff
@@ -166,115 +180,102 @@ defaultDevices.map((device) => {
...
-        .type(userData.whatsapp[randomIndex(userData.whatsapp)])
+        .type(whatsApp)
       cy.get('@whatsapp')
         .invoke('val')
         .then(($value) => {
           cy.get('@whatsapp').should('have.value', $value)
         })
-    })
+        cy.get('#btnUpdateProfile').click()
+
```
Comment (JerryHukpati) [inline cypress/integration/tests/EditProfile/amali-editprofile.spec.js:221]: remove empty line

File: cypress/integration/tests/EditProfile/amali-editprofile.spec.js [hunk 16]
```diff
@@ -166,115 +180,102 @@ defaultDevices.map((device) => {
...
     })
 
-    it('should check that country is selected', () => {
-      cy.get('[id=country]').select()
-      cy.get().click()
-    })
+    // it('should check that country is selected', () => {
+    //   cy.get('[data-cy="country"]').select()
+    //   cy.get('#btnUpdateProfile').click()
+    // })
```
Comment (JerryHukpati) [inline cypress/integration/tests/EditProfile/amali-editprofile.spec.js:194]: remove comment

File: cypress/integration/tests/EditProfile/amali-editprofile.spec.js [hunk 17]
```diff
@@ -166,115 +180,102 @@ defaultDevices.map((device) => {
...
+     cy.get('#btnUpdateProfile').click()
     })
 
-    it('should check that instagram link is vaild', () => {
-      cy.get('[id=input-133]').type(userData.instagram[randomIndex(userData.instagram)])
-      cy.get().click()
+    it('should check that instagram field is vaildated', () => {
+      cy.get('[data-cy="instagram"]').type(`${instagram}${firstName}${lastName}`)
+      cy.get('.edit-profile-btn').eq(2).contains('Save').click()
     })
```
Comment (JerryHukpati) [inline cypress/integration/tests/EditProfile/amali-editprofile.spec.js:209]: add the validations/assertions to these test cases

File: cypress/integration/tests/EditProfile/amali-editprofile.spec.js [hunk 18]
```diff
@@ -3,139 +3,153 @@
...
-      cy.get('[id="dob"]')
-        .invoke('val')
-        .then((text) => {
-          expect('@selectedDate').to.equal(text)
-        })
-    })
+    // // it('should check that date of birth is selected', () => {
+    // //   cy.get('[data-cy="dob"]').click()
+    // //  
+    // // })
```
Comment (JerryHukpati) [inline cypress/integration/tests/EditProfile/amali-editprofile.spec.js:117]: remove comment

File: cypress/integration/tests/EditProfile/amali-editprofile.spec.js [hunk 19]
```diff
@@ -3,139 +3,153 @@
...
-      cy.get('button').contains('cm')
-      cy.get('[id=input-93]')
-        .as(weight)
-        .type(userData.weight[randomIndex(userData.weight)])
-      cy.get('@weight')
+    // it('should check that prefilled email is same as sign up email', () => {
+    //   cy.get('[data-cy="email"]').as('emailField')
+    //   cy.get('@emailField').should('have.value', exist_email)
+    // })
+
```
Comment (JerryHukpati) [inline cypress/integration/tests/EditProfile/amali-editprofile.spec.js:89]: remove comment

File: cypress/integration/tests/EditProfile/amali-editprofile.spec.js [hunk 20]
```diff
@@ -3,139 +3,153 @@
...
-    it('should check that email field is non-editable', () => {
-      cy.get('[id=input-89]').as('emailField').type('testemail@gmail.com')
-      cy.get('@emailField')
-        .invoke('val')
-        .then(($text) => {
-          cy.get($text).should('not.equal', 'testemail@gmail.com')
-        })
-    })
+    it('should check that email field is non-editable and has sign up email', () => {
+
```
Comment (JerryHukpati) [inline cypress/integration/tests/EditProfile/amali-editprofile.spec.js:80]: remove empty line

File: cypress/integration/tests/EditProfile/amali-editprofile.spec.js [hunk 21]
```diff
@@ -3,139 +3,153 @@
...
   
 
-    it.only('should require full name is entered', () => {
-      cy.get('.white--text').should('contain','Full Name*').parent().next().children().first().type('ggg')
-      cy.pause()
-
-      // cy.get('input[placeholder="e.g. Mattew Alison]').clear().type('hello')
-      // cy.get('.edit-profile-btn').contains('Save').click()
+    it('should check that full name is required', () => {
+
```
Comment (JerryHukpati) [inline cypress/integration/tests/EditProfile/amali-editprofile.spec.js:62]: remove empty line

File: cypress/integration/tests/EditProfile/amali-editprofile.spec.js [hunk 22]
```diff
@@ -3,139 +3,153 @@
...
     afterEach(() => {
-      // cy.logOut()  
+      cy.location().then((loc) => {
+        if (loc.pathname === '/profile') {
+          cy.logOut()
+        } else {
+          cy.reload()
+        }
+      })
+
```
Comment (JerryHukpati) [inline cypress/integration/tests/EditProfile/amali-editprofile.spec.js:56]: remove empty line

File: cypress/integration/tests/EditProfile/amali-editprofile.spec.js [hunk 23]
```diff
@@ -3,139 +3,153 @@
...
+  address,
+  postalCode,
+  city,
+  facebook,
+  twitter,
+  instagram,
+  whatsApp,
+} = editProfileData()
+
+// defaultDevices.map((device) => {
```
Comment (JerryHukpati) [inline cypress/integration/tests/EditProfile/amali-editprofile.spec.js:32]: if you are removing the test for mobile, add a multiline comment and explain why. clean up the test by removing all commented out code

File: cypress/integration/tests/EditProfile/amali-editprofile.spec.js [hunk 24]
```diff
@@ -3,139 +3,153 @@
...
   describe(`${testSpecName} @@ Editing user profile`, () => {
-    const [w, h] = device.viewport
+    // const [w, h] = device.viewport
```
Comment (JerryHukpati) [inline cypress/integration/tests/EditProfile/amali-editprofile.spec.js:34]: refer to earlier comment on devices

File: cypress/integration/tests/EditProfile/amali-editprofile.spec.js [hunk 25]
```diff
@@ -3,139 +3,153 @@
...
 
     before(() => {
-      cy.viewport(w, h)
+      // cy.viewport(w, h)
```
Comment (JerryHukpati) [inline cypress/integration/tests/EditProfile/amali-editprofile.spec.js:37]: refer to earlier comment on devices

File: cypress/integration/tests/EditProfile/amali-editprofile.spec.js [hunk 26]
```diff
@@ -3,139 +3,153 @@
...
       cy.clearCookies({ domain: null })
       
     })
 
     beforeEach(() => {
+      // cy.viewport(w, h)
```
Comment (JerryHukpati) [inline cypress/integration/tests/EditProfile/amali-editprofile.spec.js:43]: refer to earlier comment on devices

File: cypress/fixtures/amali-create.account.data.js [hunk 27]
```diff
@@ -1,11 +1,8 @@
-import { getRandomNum } from "../integration/_helpers/helperfunc";
-
-export function newuser (){
-  
-  return  {email:  'tester' + getRandomNum() +'@yopmail.com',
-    password: 'testpassword',
-    confirm_password: 'testpassword'
+export function newUser() {
+  return {
+    email: 'aegis.tester+' +new Date().getTime().toString().slice(5, 9) + '@gmail.com',
```
Comment (JerryHukpati) [inline cypress/fixtures/amali-create.account.data.js:3]:
```suggestion
    email: 'aegis.tester+' + new Date().getTime().toString().slice(5, 9) + '@gmail.com',
```

File: cypress/fixtures/amali-checkemail.scenario.data.js [hunk 28]
```diff
@@ -1,8 +1,7 @@
-import { getRandomNum } from "../integration/_helpers/helperfunc";
 
 export const users = {
   exist_email: 'welbeckborngreat@gmail.com',
-  nonexisting_email: 'tester+' + getRandomNum() + '@gmail.com',
+  nonexisting_email: 'aegistester+' +new Date().getTime() + '@gmail.com',
```
Comment (JerryHukpati) [inline cypress/fixtures/amali-checkemail.scenario.data.js:4]:
```suggestion
  nonexisting_email: 'aegistester+' + new Date().getTime() + '@gmail.com',
```

**Review by JerryHukpati — CHANGES_REQUESTED** [review]
the edit-profile test is breaking. let's work on this together tomorrow at the office. all comments on the other changes should also be fixed

File: cypress/support/commands.js [hunk 29]
```diff
@@ -226,11 +226,21 @@ Cypress.Commands.overwrite('visit', (originalFn, url, options) => {
...
+
+Cypress.Commands.add('login', (loginEmail, loginPassword) => {
+  cy.checkEmail()
+  cy.get('#signinEmailField').should('have.value', loginEmail)
+  cy.get('[data-cy=signinPassword]').type(loginPassword)
+  cy.get('[data-cy=siginBtn]').click()
+  cy.location().should((loc) => {
+    expect(loc.pathname).to.eq('/profile')
+  })
 })
```
[the remark this answers is not in the document]
Comment (lindakumah) [reply cypress/support/commands.js:246]: We can keep this one. Felix will take the other one out?

File: cypress/fixtures/amali-create.account.data.js (hunk 27 above)
Comment (lindakumah) [reply cypress/fixtures/amali-create.account.data.js:3]: Done!

**Review by JerryHukpati — APPROVED** [review]
all good. this PR can be merged

**Pull request merged**